### PR TITLE
handle corrrelation logs when correlation enable

### PR DESCRIPTION
**Pull request by tharindu1st**

### CODE REVIEW

**Review by isudana — APPROVED** [review]

**Comment by Nirothipan** [comment]
@tharindu1st  could you please remove the .exec file

File: modules/commons/src/main/java/org/apache/synapse/commons/logger/ContextAwareLogger.java [hunk 1]
```diff
@@ -28,6 +28,15 @@
  */
 public class ContextAwareLogger {
 
+    private static Boolean correlationLoggingEnabled = false;
+
+    static {
+        String sysCorrelationStatus = System.getProperty(CorrelationConstants.CORRELATION_LOGS_SYS_PROPERTY);
+        if (sysCorrelationStatus != null) {
+            correlationLoggingEnabled = sysCorrelationStatus.equalsIgnoreCase("true");
```
Comment (Nirothipan) [inline modules/commons/src/main/java/org/apache/synapse/commons/logger/ContextAwareLogger.java:36]: without doing null check, shall we invert this comparison

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceHandler.java [hunk 2]
```diff
@@ -214,8 +214,10 @@ public void setCorrelationId(NHttpServerConnection conn) {
         } else {
             correlationId = UUID.randomUUID().toString();
             conn.getHttpRequest().setHeader(correlationHeaderName, correlationId);
+            httpContext.setAttribute(CorrelationConstants.SYSTEM_GENERATED_CORRELATION_ID, true);
         }
         httpContext.setAttribute(CorrelationConstants.CORRELATION_ID, correlationId);
+        httpContext.setAttribute(CorrelationConstants.SYSTEM_GENERATED_CORRELATION_ID, false);
```
Comment (isudana) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceHandler.java:220]: isn't this makes SYSTEM_GENERATED_CORRELATION_ID to false always?

**Review by isudana — APPROVED** [review]

**Comment by tharindu1st** [comment]
Build was failing due to connection timeout at pulling artifact

2021-04-01T23:46:53.3525754Z [ERROR] Failed to execute goal on project org.wso2.micro.integrator.management.apis: Could not resolve dependencies for project org.wso2.ei:org.wso2.micro.integrator.management.apis:bundle:4.0.0-SNAPSHOT: Could not transfer artifact org.javassist:javassist:jar:3.27.0-GA from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/org/javassist/javassist/3.27.0-GA/javassist-3.27.0-GA.jar: Connection timed out (Read failed) -> [Help 1]

**Pull request merged**